api/app/Models/Pool.php

```diff
@@ -307,6 +308,24 @@ public function getIsCompleteAttribute()
         return true;
     }
 
+    // is the assessment plan for the pool considered "complete"
+    public function getAssessmentPlanIsCompleteAttribute()
+    {
+        $pool = $this->load(['assessmentSteps', 'poolSkills']);
+
+        $planCompletionValidation = new AssessmentPlanIsCompleteValidator;
+        $validator = Validator::make($pool->toArray(),
+            $planCompletionValidation->rules(),
+            $planCompletionValidation->messages()
+        );
+
+        if ($validator->fails()) {
+            return false;
+        }
+
+        return true;
+    }
+
```

Comment:
It seems a little odd to use validation to compute a field. I think we should be reserving validation for its purpose of checking user input rather than computing an attribute :thinking:

Reply:
We've been using it for pool completeness too. 

As well, I think it makes sense as it is reusing shared logic. As in, pool completeness is required for publishing, therefore, the completion status as well validation in publishing both check the same thing. Makes sense to me to re-use rather than rewrite out the validator